sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/commands.scala

```diff
@@ -47,18 +47,38 @@ case class AnalyzeTable(tableName: String) extends RunnableCommand {
 }
 
 /**
- * Drops a table from the metastore and removes it if it is cached.
+ * Drops a table/view from the metastore and removes it if it is cached.
  */
 private[hive]
 case class DropTable(
-    tableName: String,
-    ifExists: Boolean) extends RunnableCommand {
+    tableName: TableIdentifier,
+    ifExists: Boolean,
+    isView: Boolean) extends RunnableCommand {
 
   override def run(sqlContext: SQLContext): Seq[Row] = {
     val hiveContext = sqlContext.asInstanceOf[HiveContext]
-    val ifExistsClause = if (ifExists) "IF EXISTS " else ""
+
+    // If the command DROP VIEW is to drop a table or DROP TABLE is to drop a view
+    // issue an exception.
+    val (dropTableUsingDropView, dropViewUsingDropTable) = try {
+      // If the table/view does not exist, it will throw an exception:
+      // - AnalysisException, if it is from InMemoryCatalog.
+      // - NoSuchTableException, if it is from HiveExternalCatalog
+      val catalogTable = sqlContext.sessionState.catalog.getTable(tableName)
+      (catalogTable.viewOriginalText.isEmpty && isView,
```

Comment:
Is `CatalogTable.tableType` not a better way to check if something is a view?

It also might be better to have this try statement just return a CatalogTable option, and handle the potential (lack of) existence and further steps later on.